script.skin.helper.colorpicker/default.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import xbmc,xbmcgui,xbmcaddon
+import resources.lib.ColorPicker as colorpicker
+
+ADDON = xbmcaddon.Addon()
+ADDON_ID = ADDON.getAddonInfo('id').decode("utf-8")
+ADDON_PATH = ADDON.getAddonInfo('path').decode("utf-8")
+
+def getParams():
+    #extract the params from the called script path
+    params = {}
+    for arg in sys.argv:
+        arg = arg.decode("utf-8")
+        if arg == 'script.skin.helper.colorpicker' or arg == 'default.py':
+            continue
+        elif "=" in arg:
+            paramname = arg.split('=')[0]
+            paramvalue = arg.replace(paramname+"=","")
+            params[paramname] = paramvalue
+            params[paramname.upper()] = paramvalue
+    return params
+    
+def waitForSkinShortcutsWindow():
+    #wait untill skinshortcuts is active window (because of any animations that may have been applied)
+    for i in range(40):
+        if not (xbmc.getCondVisibility("Window.IsActive(DialogSelect.xml) | Window.IsActive(script-skin_helper_service-ColorPicker.xml) | Window.IsActive(DialogKeyboard.xml)")):
+            break
+        else: xbmc.sleep(100)
+    
+
+#MAIN ENTRY POINT    
+params = getParams()
+
+if params and params.get("ACTION","").upper() == "COLORPICKER":
```

Comment:
you can probably leave out the action parameter now? (since it only supports one action)